dash/include/dash/coarray/Utils.h

```diff
@@ -0,0 +1,212 @@
+#ifndef DASH__COARRAY_UTILS_H__
+#define DASH__COARRAY_UTILS_H__
+
+#include <dash/Types.h>
+
+/**
+ * \defgroup  DashCoarrayLib  Coarray Runtime Interface
+ * 
+ * \ingroup DashCoarrayConcept
+ *
+ * Functions of the \c dash::coarray extension to provide an interface similar to
+ * CAF 2008
+ */
+
+namespace dash {
+namespace coarray {
+
+/**
+  * Shortcut to query the global unit ID of the calling unit.
+  *
+  * \return The unit ID of the calling unit relative to \ref dash::Team::All
+  *
+  * \sa dash::myid()
+  *
+  * \ingroup DashCoarrayLib
+  */
+inline dash::global_unit_t this_image() {
+  return dash::myid();
+}
+
+/**
+  * Return the number of units in the global team.
+  *
+  * \return The number of units available.
+  *         -1 if DASH is not initialized (anymore).
+  *
+  * \sa dash::size()
+  *
+  * \ingroup DashCoarrayLib
+  */
+inline ssize_t num_images() {
+  return dash::size();
+}
+
+/**
+ * blocks until all units reach this statement. This statement does not 
+ * imply a flush. If a flush is required, use the corresponding
+ * \c dash::Coarray::sync_all() method of Coarray.
+ * 
+ * \sa dash::coarray::sync_images()
+ * 
+ * \ingroup DashCoarrayLib
+ */
+inline void sync_all(){
+  dash::barrier();
+}
+
+/**
+ * Blocks until all selected units reach this statement. This statement does
+ * not imply a flush. If a flush is required, use the \c sync_all() method of
+ * the Coarray
+ * 
+ * \note If possible use \c sync_all() or \c Coevent for performance reasons.
+ *       \c sync_images() is implemented using two-sided operations based on 
+ *       the implementation of \c MPI_Barrier() in OpenMPI
+ * 
+ * \sa dash::coarray::sync_all()
+ * 
+ * \ingroup DashCoarrayLib
+ */
+template<typename Container>
+inline void sync_images(const Container & image_ids){    
```

Comment:
It might be more efficient to implement this functionality in DART (using MPI non-blocking communication). I will keep that in mind.

Reply:
Well ... It might be and DART is the correct place for such things, however @fuerlinger voted for removing this function as it totally contradicts the team concept.

Reply:
If this is a CAF/Coarray concept maybe we should leave it in (it doesn't sound like it would scale to the extremes though). I guess you already had sufficient discussions on this :)